cheatsheets/File_Upload_Cheat_Sheet.md

```diff
@@ -103,8 +103,23 @@ Based on the expected type, special file content validation can be applied:
 
 - For **images**, applying image rewriting techniques destroys any kind of malicious content injected in an image; this could be done through [randomization](https://security.stackexchange.com/a/8625/118367).
 - For **Microsoft documents**, the usage of [Apache POI](https://poi.apache.org/) helps validating the uploaded documents.
+- For **PDF documents**, the usage of [Apache PDFBox](https://pdfbox.apache.org/) helps validating the uploaded documents.
 - **ZIP files** are not recommended since they can contain all types of files, and the attack vectors pertaining to them are numerous.
 
+It is possible to "hide" a malicious file in a document or image, by adding it to the end of the source file, like this:
+
+```shell
+$ file safe-document.pdf
+safe-document.pdf: PDF document, version 1.4
+$ file malicious-file.exe
+malicious-file.exe: PE32+ executable (console) x86-64, for MS Windows
+$ cat safe-document.pdf malicious-file.exe > malicious-document.pdf
+$ file malicious-document.pdf
+malicious-document.pdf: PDF document, version 1.4
+```
+
+Therefore, it is recommended like mentioned above for images, to apply document rewriting techniques to destroys any kind of malicious content embedded.
```

Comment:
Why the output of 'file' in this case shows what it does makes perfect sense, but this is really only a problem if your OS (or perhaps your PDF viewer) is going to attempt to execute the 'malicious-file.exe' part, which seems a stretch even for Windows OS.  I have seen malicious content injected into PDFs via JavaScript links and leveraging bugs in common PDF viewers such as Adobe Acrobat Reader, but it was never anything as simple as this.

So, so you have a documented case where something was exploited using this simple approach? If so, I'd like a reference to it.

Reply:
It is critical that untrusted uploaded files staged in public directories are assigned file permissions of read only at the OS level. That is my only execution concern overall.